pkg/executor/sortexec/sort_partition.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sortexec
+
+import (
+	"fmt"
+	"sort"
+	"unsafe"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/pkg/types"
+	"github.com/pingcap/tidb/pkg/util/chunk"
+	"github.com/pingcap/tidb/pkg/util/disk"
+	"github.com/pingcap/tidb/pkg/util/memory"
+)
+
+var errSpillIsTriggered = errors.New("can not add because spill has been triggered")
+var errSpillEmptyChunk = errors.New("can not spill empty chunk to disk")
+
+const rowPtrSize = int(unsafe.Sizeof(chunk.RowPtr{}))
+
+// It should be const, but we need to modify it for test.
```

Comment:
How about being a member variable of `sortPartition`?

Reply:
> How about being a member variable of `sortPartition`?

I think it's better to be global variable as all `sortPartition` share the same `spillChunkSize`.